horovod/tensorflow/__init__.py

```diff
@@ -237,3 +247,83 @@ def get_slot_names(self, *args, **kwargs):
     def variables(self, *args, **kwargs):
         """Calls this same method on the underlying optimizer."""
         return self._optimizer.variables(*args, **kwargs)
+
+
+class _DistributedGradientTape(tf.GradientTape):
```

Comment:
Older versions of TensorFlow don't have GradientTape, can you wrap this and following declaration into `if hasattr(tf, 'GradientTape')` to fix the tests?